static/panes/executor.js

```diff
@@ -125,17 +125,18 @@ Executor.prototype.compilerIsVisible = function (compiler) {
 };
 
 Executor.prototype.getEditorIdBySourcefile = function (sourcefile) {
-    if (this.sourceTreeId) {
```

Comment:
sourcefile being empty is always a questionable situation, but if sourcefile is empty, it does not follow that sourceEditorId is set. sourceEditorId is only set if the Executor was based on an editor. If the executor was made from the tree, the tree is in change of editors because the tree knows if the editor is active or not (because you can close the editors as long as they have a name and are saved to the tree)

Reply:
so we're missing a version now where sourcefile is not set but this.sourceTreeId is set

Reply:
Instead of copying compiler.js, I've done something similar to what output.ts does, but I needed to modify utils::parseOutput, which makes me wonder I again am not onto the right solution. wdyt?

Reply:
The whole utils parseOutput is very inconsistently used and has issues already, I wouldn't worry about it too much, we'll get to it some other time.
I think this is Ok for now.